### PR TITLE
Benchmark Type-specific Encoding 

### PR DESCRIPTION
**Pull request by lawben**
This PR adds type-specific encodings, i.e. all `int` columns are FrameOfReference-encoded.

```

{
  "default": {
    "encoding": <ENCODING_TYPE_STRING>,               // required
    "compression": <VECTOR_COMPRESSION_TYPE_STRING>,  // optional
  },

  "type": {
    <DATA_TPYE>: {
      "encoding": <ENCODING_TYPE_STRING>,
      "compression": <VECTOR_COMPRESSION_TYPE_STRING>
    },
    <DATA_TPYE>: {
      "encoding": <ENCODING_TYPE_STRING>
    }
  },
  ...
}
```

### CODE REVIEW

File: src/benchmarklib/benchmark_utils.hpp [hunk 1]
```diff
@@ -25,7 +25,12 @@ using TimePoint = std::chrono::high_resolution_clock::time_point;
 using NamedQuery = std::pair<std::string, std::string>;
 using NamedQueries = std::vector<NamedQuery>;
 
+<<<<<<< HEAD
```
Comment (mrks) [inline src/benchmarklib/benchmark_utils.hpp]: still broken

Comment (lawben) [reply src/benchmarklib/benchmark_utils.hpp]: jup, I accidentally open the PR while I wasn't ready yet :/ give me a sec

Comment (lawben) [reply src/benchmarklib/benchmark_utils.hpp]: @mrks should be ready now

**Comment by mrks** [comment]
Could you please add some words here: https://github.com/hyrise/hyrise/wiki/GeneralPurposeBenchmarkRunner

**Comment by lawben** [comment]
@mrks I added some stuff. is that okay or is something missing?

**Comment by mrks** [comment]
"some" :D 
Thanks!

One more question: When outputting the result JSON, do we already include the encoding configuration. I am thinking about the case where we want to find the best encoding e.g., for TPC-H. For that, it would be good to be able to distinguish different runs.

**Comment by lawben** [comment]
yes, we do. current example output looks like this:

```json
{
  "benchmarks": [
    {
      "avg_real_time_per_iteration": 269733964,
      "items_per_second": 3.70735645294189,
      "iterations": 19,
      "name": "TPC-H 1",
      "time_unit": "ns"
    },
    {
      "avg_real_time_per_iteration": 603366741,
      "items_per_second": 1.65736675262451,
      "iterations": 9,
      "name": "TPC-H 9",
      "time_unit": "ns"
    }
  ],
  "context": {
    "GIT-HASH": "f7051ca6011ccc3d26ce33e296b2cad0813570bf-dirty",
    "benchmark_mode": "IndividualQueries",
    "build_type": "debug",
    "chunk_size": 10000,
    "date": "2018-06-25 11:46:07",
    "encoding": {
      "custom": {
        "customer": {
          "c_custkey": {
            "compression": "Fixed-size byte-aligned",
            "encoding": "Dictionary"
          },
          "c_nationkey": {
            "encoding": "RunLength"
          }
        },
        "lineitem": {
          "l_partkey": {
            "encoding": "FrameOfReference"
          }
        }
      },
      "default": {
        "encoding": "Dictionary"
      },
      "type": {
        "int": {
          "encoding": "FrameOfReference"
        },
        "long": {
          "compression": "Fixed-size byte-aligned",
          "encoding": "FrameOfReference"
        },
        "string": {
          "compression": "SIMD-BP128",
          "encoding": "Dictionary"
        }
      }
    },
    "max_duration (s)": 5,
    "max_runs": 1000,
    "output_file_path": "stdout",
    "scale_factor": 0.00100000004749745,
    "using_mvcc": false,
    "using_scheduler": true,
    "using_visualization": false,
    "verbose": true
  }
```

**Comment by mrks** [comment]
Great!

**Pull request merged**